internal/keyrotation/config.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keyrotation
+
+import (
+	"time"
+
+	"github.com/google/exposure-notifications-server/internal/database"
+	"github.com/google/exposure-notifications-server/internal/observability"
+	"github.com/google/exposure-notifications-server/internal/publish"
+	"github.com/google/exposure-notifications-server/internal/setup"
+	"github.com/google/exposure-notifications-server/internal/storage"
+	"github.com/google/exposure-notifications-server/pkg/secrets"
+)
+
+// Compile-time check to assert this config matches requirements.
+var _ setup.BlobstoreConfigProvider = (*Config)(nil)
+var _ setup.DatabaseConfigProvider = (*Config)(nil)
+var _ setup.SecretManagerConfigProvider = (*Config)(nil)
+var _ setup.ObservabilityExporterConfigProvider = (*Config)(nil)
+
+// Config represents the configuration and associated environment variables for
+// the key rotation components.
+type Config struct {
+	// TODO(whaught): Copied. Not sure if we will actually need all of these.
+	Database              database.Config
+	SecretManager         secrets.Config
+	Storage               storage.Config
+	ObservabilityExporter observability.Config
+
+	RevisionToken publish.RevisionTokenConfig
+
+	Port           string        `env:"PORT, default=8080"`
+	RotationPeriod time.Duration `env:"ROTATION_PERIOD, default=14d"`
+	MinKeys        int           `env:"MIN_KEYS, default=2"`
```

Comment:
it's no Min # of keys that's needed - it's a min effective time period. 

We need to make sure we have 15 days of coverage - theoretically 14d, but timezones are hard, so we usually round up to 15d.